### PR TITLE
fix: restructure resend email code to send email code

### LINKED ISSUE #9096: `hasCode` and `sendCode` requests
### Preflight Checklist

- [x] I could not find a solution in the existing issues, docs, nor discussions
- [x] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Describe your problem

When verifying user emails, we have a scenario where a user never got a verification email. To react to such a state we need to send a mail whenever the verification page is shown to the user. We therefore need a `hasCode` and `sendCode` method.

### Describe your ideal solution

Provide methods `hasCode` and `sendCode` on the user service

### Version

_No response_

### Environment

ZITADEL Cloud

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by stebenz**
# Which Problems Are Solved

There is currently no endpoint to send an email code for verification of the email if you don't change the email itself.

# How the Problems Are Solved

Endpoint HasEmailCode to get the information that an email code is existing, used by the new login.
Endpoint SendEmailCode, if no code is existing to replace ResendEmailCode as there is a check that a code has to be there, before it can be resend.

# Additional Changes

None

# Additional Context

Closes #9096

### CODE REVIEW

File: internal/command/user_v2_email_test.go [hunk 1]
```diff
@@ -760,6 +931,85 @@ func TestCommands_ResendUserEmailReturnCode(t *testing.T) {
...
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &Commands{
+				eventstore:      tt.fields.eventstore,
+				checkPermission: tt.fields.checkPermission,
+			}
+			_, err := c.SendUserEmailReturnCode(context.Background(), tt.args.userID, crypto.CreateMockEncryptionAlg(gomock.NewController(t)))
+			require.ErrorIs(t, err, tt.wantErr)
+			// successful cases are tested in TestCommands_resendUserEmailCodeWithGenerator
```
Comment (adlerhurst) [inline internal/command/user_v2_email_test.go:1008]: I cannot find the tests mentioned.
```suggestion
			// successful cases are tested in TestCommands_sendUserEmailCodeWithGenerator
```

File: internal/command/user_v2_email_test.go [hunk 2]
```diff
@@ -643,6 +722,98 @@ func TestCommands_ResendUserEmailCodeURLTemplate(t *testing.T) {
...
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &Commands{
+				eventstore:      tt.fields.eventstore,
+				checkPermission: tt.fields.checkPermission,
+			}
+			_, err := c.SendUserEmailCodeURLTemplate(context.Background(), tt.args.userID, crypto.CreateMockEncryptionAlg(gomock.NewController(t)), tt.args.urlTmpl)
+			require.ErrorIs(t, err, tt.wantErr)
+			// successful cases are tested in TestCommands_resendUserEmailCodeWithGenerator
```
Comment (adlerhurst) [inline internal/command/user_v2_email_test.go:812]: I cannot find the tests mentioned.
```suggestion
			// successful cases are tested in TestCommands_sendUserEmailCodeWithGenerator
```

File: internal/command/user_v2_email_test.go [hunk 3]
```diff
@@ -512,6 +512,85 @@ func TestCommands_ResendUserEmailCode(t *testing.T) {
...
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &Commands{
+				eventstore:      tt.fields.eventstore,
+				checkPermission: tt.fields.checkPermission,
+			}
+			_, err := c.SendUserEmailCode(context.Background(), tt.args.userID, crypto.CreateMockEncryptionAlg(gomock.NewController(t)))
+			require.ErrorIs(t, err, tt.wantErr)
+			// successful cases are tested in TestCommands_changeUserEmailWithGenerator
```
Comment (adlerhurst) [inline internal/command/user_v2_email_test.go:589]: I cannot find the tests mentioned.
```suggestion
			// successful cases are tested in TestCommands_sendUserEmailCodeWithGenerator
```

Comment (adlerhurst) [reply internal/command/user_v2_email_test.go:589]: I think same applies to line [720](https://gitkraken.dev/link/dnNjb2RlOi8vZWFtb2Rpby5naXRsZW5zL2xpbmsvci82NjFiY2ZiZGVjZGNjNzliNGEyOGU1YmIyNTJhNTgwZGU0YmE1MzJjL2YvaW50ZXJuYWwvY29tbWFuZC91c2VyX3YyX2VtYWlsX3Rlc3QuZ28%2FdXJsPWdpdCU0MGdpdGh1Yi5jb20lM0F6aXRhZGVsJTJGeml0YWRlbC5naXQmbGluZXM9NzIw?origin=gitlens)

**Review by adlerhurst — CHANGES_REQUESTED** [review]
code looks good, only 4 comments which are wrong

**Comment by adlerhurst** [comment]
please also have a look at the copilot review

**Comment by stebenz** [comment]
> please also have a look at the copilot review

Corrected the wrongly mentioned functions in the comments.
Had a look at the copilot review, can change the text, but as it is a oneOf and 'nil' is also checked, default will never happen through an API call. Should we leave this be for now or introduce unit testing for all oneOf mappings in the API?

**Review by adlerhurst — APPROVED** [review]

**Pull request merged**